sematic/api_client.py

```diff
@@ -71,13 +68,15 @@ def get_artifact_value_by_id(
     Any
         The value of the requiested artifact.
     """
-    # TODO: Store storage type on artifact
-    if storage is None:
-        storage = S3Storage()
-
     artifact = _get_artifact(artifact_id)
 
-    return get_artifact_value(artifact, storage)
+    return get_artifact_value(artifact)
```

Comment:
Implementation-wise, `artifact = _get_artifact(artifact_id)` might be an unnecessary round-trip --- we incurred a heavy read on the full payload of artifact, after we toss it to `get_artifact_value()`, it only needs its id. 

Why not just do
```
    payload = _get_artifact_bytes(artifact_id)
    return deserialize_artifact_value(artifact, payload)
```

`get_artifact_value` should call this one, not the other way around.

Reply:
Unfortunately, we need `artifact` for `deserialize_artifact_value`, since it contains the type serialization, which will inform how to deserialize the payload.

I don't believe there is unnecessary rount-trip:
- `_get_artifact` fetches the artifact record from the DB. It does not contain the full artifact serialization.
- `_get_artifact_bytes` fetches the artifact payload
- `deserialize_artifact_value` does not fetch anything

Reply:
I see. Thanks for the explanation.